app/controllers/user_notifications_controller.rb

```diff
@@ -21,4 +21,28 @@ def delete
     ).destroy!
     render nothing: true
   end
+
+  def unsubscribe
+    user_notif = UserNotification.find_by(token: params[:token])
+    if user_notif.present?
+      Notification.find_each do |n|
+        notif = UserNotification.find_by(
+          user_id: user_notif.user_id, notification_id: n.id
+        )
+        notif.destroy! if notif.present?
+      end
+    end
+    flash.now[:alert] = 'Anda telah unsubscribe email KTOM'
+  end
+
+  def stop_this_notification
+    user_notif = UserNotification.find_by(token: params[:token])
+    if user_notif.present?
+      notif = UserNotification.find_by(
+        user_id: user_notif.user_id, notification_id: params[:notification_id]
```

Comment:
If the token already uniquely identifies a `UserNotification` object, what is the use of this `notification_id`?